spanner/session.go

```diff
@@ -1462,6 +1666,11 @@ func (hc *healthChecker) markDone(s *session) {
 // healthCheck checks the health of the session and pings it if needed.
 func (hc *healthChecker) healthCheck(s *session) {
 	defer hc.markDone(s)
+	// If the session is multiplexed and has been idle for more than 7 days,
+	if s.isMultiplexed && s.createTime.Add(multiplexSessionIdleTime).Before(time.Now()) {
```

Comment:
This should not be part of the standard health-check feature for regular sessions. This method is called by a worker that looks for sessions that need a ping. That logic is not the right logic that should determine whether we need to check whether the multiplexed session should be refreshed. It is probably better to put it in a separate method that only takes care of refreshing multiplexed sessions.
Also, we should ensure that:
1. The method (by default) does not need to be called very often. We are checking whether the session is more than 7 days old. That is something that can be checked at an interval of 10 minutes or something like that.
2. We should only have one `createMultiplexedSession` call in-flight at any time.

Reply:
1) Created separate worker `multiplexSessionWorker` to refresh session with 10 minute interval.
2) Added test `TestClient_MultiplexedSession` to validate one in-flight req `createMultiplexedSession` at any time.